pkgs/applications/networking/instant-messengers/telegram/tdesktop/tg_owt.nix

```diff
@@ -1,30 +1,42 @@
-{ lib, stdenv, fetchFromGitHub, pkg-config, cmake, ninja, yasm
+{ lib, stdenv, fetchFromGitHub, fetchpatch
+, pkg-config, cmake, ninja, yasm
 , libjpeg, openssl, libopus, ffmpeg, alsa-lib, libpulseaudio, protobuf
-, xorg, libXtst
+, xorg, libXtst, libXcomposite, libXdamage, libXext, libXrender, libXrandr
+, glib, abseil-cpp, pcre, util-linuxMinimal, libselinux, libsepol, pipewire
 }:
 
-let
-  rev = "2d804d2c9c5d05324c8ab22f2e6ff8306521b3c3";
-  sha256 = "0kz0i381iwsgcc3yzsq7njx3gkqja4bb9fsgc24vhg0md540qhyn";
-
-in stdenv.mkDerivation {
+stdenv.mkDerivation {
   pname = "tg_owt";
-  version = "git-${rev}";
+  version = "unstable-2021-06-17";
 
   src = fetchFromGitHub {
     owner = "desktop-app";
     repo = "tg_owt";
-    inherit rev sha256;
+    rev = "f03ef05abf665437649a4f71886db1343590e862";
+    sha256 = "0s6ajw52b95lcq4mn6lv8gj6lhv62bvwjh43w7az2k5pbm14v7vv";
     fetchSubmodules = true;
   };
 
+  patches = [
+    # Our libXtst seems broken:
+    # /nix/store/rd3swxwmzjgjvwhz8svyc8ghc0brq293-libXtst-1.2.3/include/X11/extensions/XTest.h:32:10: fatal error: X11/extensions/XInput.h: No such file or directory
```

Comment:
That looks like you need to add libXi

Reply:
Yes, `xorg.libXi.dev` provides the necessary header but I doubted that it'll get picked up automatically (requires the correct include path). However, I did just look into `xtst.pc` and it contains `Requires.private: x11 xext xextproto xi` (my bad :o) so it should actually work, I think (my pkg-config knowledge is pretty limited). I thought it was broken because `libXtst` should handle propagated dependencies but I forgot that `tg_owt` uses static linking (which I'm not used to) and `libXi` is only required because of that.

Reply:
Implemented in #128295.